presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeIcebergTpcdsWithPositionalDeletes.java

```diff
@@ -0,0 +1,265 @@
+                "                           FROM   date_dim " +
+                "                           WHERE  d_date BETWEEN date'2002-11-14' AND date'2002-11-21') ", 2002L);
+        deletedRowsMapBuilder.put("inventory", 6006L);
+
+        assertUpdateExpected(session, "DELETE FROM catalog_returns " +
+                        "WHERE  cr_order_number IN (SELECT cs_order_number " +
+                        "                           FROM   catalog_sales, " +
+                        "                                  date_dim " +
+                        "                           WHERE  cs_sold_date_sk = d_date_sk " +
+                        "                                  AND d_date BETWEEN date'2000-05-20' AND " +
+                        "                                                     date'2000-05-21') ",
+                7L);
+
+        assertUpdateExpected(session, "DELETE FROM catalog_returns " +
+                        "WHERE  cr_order_number IN (SELECT cs_order_number " +
+                        "                           FROM   catalog_sales, " +
+                        "                                  date_dim " +
+                        "                           WHERE  cs_sold_date_sk = d_date_sk " +
+                        "                                  AND d_date BETWEEN date'1999-09-18' AND " +
+                        "                                                     date'1999-09-19') ",
+                12L);
+
+        assertUpdateExpected(session, "DELETE FROM catalog_returns " +
+                        "WHERE  cr_order_number IN (SELECT cs_order_number " +
+                        "                           FROM   catalog_sales, " +
+                        "                                  date_dim " +
+                        "                           WHERE  cs_sold_date_sk = d_date_sk " +
+                        "                                  AND d_date BETWEEN date'2002-11-12' AND " +
+                        "                                                     date'2002-11-13') ",
+                15L);
+        deletedRowsMapBuilder.put("catalog_returns", 34L);
+
+        assertUpdateExpected(session, "DELETE FROM catalog_sales " +
+                        "WHERE  cs_sold_date_sk >= (SELECT Min(d_date_sk) " +
+                        "                           FROM   date_dim " +
+                        "                           WHERE  d_date BETWEEN date'2000-05-20' AND date'2000-05-21') " +
+                        "       AND cs_sold_date_sk <= (SELECT Max(d_date_sk) " +
+                        "                               FROM   date_dim " +
+                        "                               WHERE  d_date BETWEEN date'2000-05-20' AND " +
+                        "                                                     date'2000-05-21') ",
+                54L);
+
+        assertUpdateExpected(session, "DELETE FROM catalog_sales " +
```

Comment:
In each delete function section, put the _sales ones in front of the _returns to match the spec

Reply:
It seems delete on _returns should be run first and then _sales for the selected date range. If we delete the rows from _sales first, then _returns will not have anything to delete from the same date range since the _returns table DELETE query uses _sales join with date_dim.